crmsh/report/utils.py

```diff
@@ -362,7 +362,29 @@ def get_pkg_mgr() -> str:
     return ""
 
 
-def get_timestamp(line: str) -> float:
+def get_timestamp_from_time_line(time_line: str, stamp_type: str, log_file: str):
+    timestamp = crmutils.parse_to_timestamp(time_line, quiet=True)
+    # got a timestamp in the future
+    if timestamp and timestamp > datetime.datetime.now().timestamp() and stamp_type == "syslog":
+        # syslog doesn't have year info, so we need to guess it
+        mtime = os.path.getmtime(log_file)
+        mtime = datetime.datetime.fromtimestamp(mtime)
+        this_year = datetime.datetime.now().year
```

Comment:
Instead of calling `datatime.now()` repeatly, the datetime instance can be cached as a local variable.